### ISSUE TITLE
Odd use of trim, kind of works by accident

### ISSUE THREAD

**Issue by peterspicer-catalyst** [davidpesce/moodle-logstore_xapi#184]
**Description**
- In src/loader/lrs.php, there is a line that reads `$nostatements = trim($endpoint, 'statements');` - while this "works", it does not work by way of removing the word 'statements' from the end of the URL, it trims any leading or trailing instances of the s, t, a, e, m, or n characters as trim carries out.

**Version**
- relevant file last changed in caea2c2, master branch on commit 57a05cb, version 3.0.1

**Steps to reproduce the bug**
1. Enter an LRS URL that doesn't end in a / but ends in one or more of any of the letters s, t, a, e, m or n. (Arbitrary example: `https://myprivatehost/lrs`)
2. The URL may now be malformed when trying to send to it. (In the example, `https://myprivatehost/lr`)

**Expected behaviour**
- load_transormed_events_to_lrs should be attempting to remove only `/statements` from the end of the URL if found.

**Actual behaviour**
- The function can end up malforming URLs to then try to send data to.

**Additional information**
- The easiest fix that comes to mind is to replace correct_endpoint as follows:

```
function correct_endpoint($endpoint) {
    if (substr($endpoint, -11) == "/statements") {
        $endpoint = substr($endpoint, 0, -11);
    }
    return rtrim($endpoint, '/');
}
```

This way you only remove /statements from the end of the URL if it legitimately ends with that, and right-trim (only) trailing / if any.

src/loader/lrs.php also contains the function load_transormed_events_to_lrs, which probably should be renamed to load_transformed_events_to_lrs sometime.

**Comment by caperneoignis** [comment]
Interesting. By the way, that is a majestic beard that one can be proud of. You are correct though, That does seem to be the easiest way, and the one least likely to remove characters it shouldn't.

**Comment by ryasmi** [comment]
Thanks for reporting the issue with a great explanation and taking the time to propose a solution. This issue will be fixed when #187 is merged 👍
